scipy/signal/tests/test_splines.py

```diff
@@ -409,7 +412,11 @@ def test_symiir2_values(self, dtyp, xp):
         # In that respect, sosfilt is likely doing a better job.
         xp_assert_close(res, exp_res, atol=2e-6, check_dtype=False)
 
-        s = s + 1j*s
+        I1 = xp.asarray(
+            1 + 1j, dtype=xp.complex64 if s.dtype == xp.float32 else xp.complex128
+        )
+        s = s * I1
```

Comment:
nit: this is the same for all array types, right?
```suggestion
        s *= I1
```

Reply:
No idea if this works on jax, and it's not even worth checking IMO. This PR will be happily reverted as soon as the array api revision includes complex scalar support (hopefully 2024.12, hopefully soon).